Libraries/Image/AssetSourceResolver.js

```diff
@@ -71,6 +71,15 @@ class AssetSourceResolver {
     return !!(this.jsbundleUrl && this.jsbundleUrl.startsWith('file://'));
   }
 
+  isCatalogAsset(): boolean {
+    return (
+      this.asset.__packager_asset &&
```

Comment:
What if apps didn't upgrade their build scripts to asset catalogs? What about Mac Catalyst apps?

Reply:
This is definetly the main danger about this change. It does require updating the app to use asset catalogs to work, as I don't see a simple way to support falling back to looking in the app bundle if the asset doesn't exist in the catalog.

Metro doesn't really have a way to know about how assets are bundled since this is handled by the CLI in release builds. The only way I could think off is add some fallback in the iOS code to check if the asset exists in the catalog, but that seems to add complexity and probably a large perf hit if none of the assets are in the catalog, so I figured it might be better to just not support app bundle assets anymore.

I'm open to suggestions on how we could improve the migration path, since I couldn't really find a perfect solution here.

Reply:
I think that's a pretty big breaking change.

Instead, could we do the mapping from path to resource identifier on the native side, so we have the option to fallback?

I'll try to do some more tests internally, but we should definitely call this out in the changelog, and do more testing of what happens in apps not using asset catalogs.

Reply:
Ok, I'll have a look at implementing this on the native side

Reply:
Also good point about catalyst apps, will test this also. Do you know if it is possible to run RN tester as a Catalyst app?

Reply:
Currently if the assets are not in the catalog the image will be blank in production builds, which is far from ideal as it can only be catched by manual testing.

I guess we could log a warning when doing the fallback, but that might also be hard to catch since it would only be in release builds. Hard error could also be a possibility as this should be very obvious but def makes it more of a breaking change.

Reply:
It may require some updates to the xcode project, but RN Tester should be supported as a catalyst app.